UnityProject/Assets/Scripts/Items/Devices/AccessRemote.cs

```diff
@@ -0,0 +1,135 @@
+				return;
+			}
+
+			if (departmentSprite == null)
+			{
+				Logger.LogWarning("[AccessRemote] - No department sprite found, using default sprite instead. (default sprite could be blank however!)");
+				return;
+			}
+			spriteHandler.SetSpriteSO(departmentSprite);
+		}
+
+		private enum AccessRemoteState
+		{
+			Open,
+			Bolts,
+			Emergency
+		}
+
+		public void ServerPerformInteraction(HandActivate interaction)
+		{
+			switch (currentState)
+			{
+				case AccessRemoteState.Open:
+					SyncCurrentRemoteState(currentState, AccessRemoteState.Bolts);
+					break;
+				case AccessRemoteState.Bolts:
+					SyncCurrentRemoteState(currentState, AccessRemoteState.Emergency);
+					break;
+				case AccessRemoteState.Emergency:
+					SyncCurrentRemoteState(currentState, AccessRemoteState.Open);
+					break;
+			}
+			Chat.AddExamineMsg(interaction.Performer, $"Remote mode is set to: {currentState.ToString()}.");
+		}
+
+		public bool WillInteract(HandApply interaction, NetworkSide side)
+		{
+			if (interaction.IsHighlight || interaction.IsAltClick)
+				return false;
+			if (Validations.HasComponent<DoorMasterController>(interaction.TargetObject) == false)
+				return false;
+
+			return true;
+		}
+
+		public void ServerPerformInteraction(HandApply interaction)
+		{
+			DoorMasterController doorController = interaction.TargetObject.GetComponent<DoorMasterController>();
+			if(doorController == null) return;
+			AccessModule accessModule = interaction.TargetObject.GetComponentInChildren<AccessModule>();
+			Chat.AddExamineMsg(interaction.Performer, $"You use access remote on: {doorController.gameObject.ExpensiveName()}");
+			if (accessModule != null && accessModule.ProcessCheckAccess(access) == false)
+			{
+				Chat.AddExamineMsg(interaction.Performer, "This remote does not contain the required access.");
+				return;
+			}
+
+			switch (currentState)
+			{
+				case AccessRemoteState.Open:
+					TryOpenDoor(doorController, interaction.Performer);
+					break;
+				case AccessRemoteState.Emergency:
+					if (accessModule == null)
+					{
+						Chat.AddExamineMsg(interaction.Performer, $"{doorController.gameObject.ExpensiveName()} has no access module!");
+						return;
+					}
+					accessModule.ToggleAuthorizationBypassState();
+					break;
+				case AccessRemoteState.Bolts:
+					BoltsModule boltsModule = interaction.TargetObject.GetComponentInChildren<BoltsModule>();
+					if (boltsModule == null)
+					{
+						Chat.AddExamineMsg(interaction.Performer, $"{doorController.gameObject.ExpensiveName()} has no bolts module!");
+						return;
+					}
+					boltsModule.ToggleBolts();
+					break;
+				default:
+					TryOpenDoor(doorController, interaction.Performer);
+					break;
+			}
+		}
+
+		private void SyncCurrentRemoteState(AccessRemoteState oldState, AccessRemoteState newState)
+		{
+			//(Max) - Why do we need to keep track of the old state?
```

Comment:
The oldState parameter is necessary for Mirror

Reply:
yes but why do we need it? `[SyncVar]` is good enough alone

Reply:
true, so can be removed